.github/workflows/test_on_push.yml

```diff
@@ -38,11 +38,6 @@ jobs:
       matrix:
         os: [ubuntu-latest, macos-12, windows-latest]
         python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
```

Comment:
On the below line, could you exclude Ubuntu with Python 3.12 again which you removed? It creates a redundant job otherwise

Reply:
I removed it because it was easy to forget if we change the coverage test. I figured one redundant test was better than potentially missing a test. It also assumes that coverage tests will always be the same as this test

Reply:
Let's still exclude it because it is one more unneeded job after all. You can move the coverage job to the same job if you want (but the reason why we kept it separate was because of being able to alter the `name:`

Reply:
I will make adjustments for this when I do the JAX changes. I think getting rid of the exclude is still a good thing since it prevents an error in the future. It is a good design principle to only have a single point of change, so if we change the coverage version then we should not have to remember to un-exclude a different test

Reply:
Sure, resolving this comment for now

Reply:
I'll prefer keeping the `exclude` block too, will save us CI time and space. The coverage job has been a separate job for quite some time now (~3 years) and we made it a distinct job in order to save CI time. This should go away anyways with the GSoC `pytest` project, so we should keep it as it for now.

Reply:
I still strongly disagree with this skipped test, but I put it back. For the very reason that when I put it back I had to go search to find out what the coverage was currently using. It is not a good approach

Reply:
We'll make a note about this as something to be fixed during the summer